File: 4005-maximum-total-subarray-value-i/maximum-total-subarray-value-i.go
func maxTotalValue(nums []int, k int) int64 {
	mx, mn := math.MinInt, math.MaxInt

	for _, num := range nums {
		mx = max(mx, num)
		mn = min(mn, num)
	}

	return int64(mx-mn) * int64(k)
}

File: 4005-maximum-total-subarray-value-i/maximum-total-subarray-value-i.cs
public class Solution {
    public long MaxTotalValue(int[] nums, int k) {
        int mn = nums[0], mx = nums[0];

        foreach (int num in nums) {
            mn = Math.Min(mn, num);
            mx = Math.Max(mx, num);
        }

        return (long) (mx - mn) * k;
    }
}